internal/dataplane/translator/backendref.go

```diff
@@ -23,42 +34,65 @@ func backendRefsToKongStateBackends(
 	for _, backendRef := range backendRefs {
 		logger := loggerForBackendRef(logger, route, backendRef)
 
-		if util.IsBackendRefGroupKindSupported(
-			backendRef.Group,
-			backendRef.Kind,
-		) && newRefChecker(backendRef).IsRefAllowedByGrant(allowed) {
-			port := int32(-1)
-			if backendRef.Port != nil {
-				port = int32(*backendRef.Port)
-			}
-			namespace := route.GetNamespace()
-			if backendRef.Namespace != nil {
-				namespace = string(*backendRef.Namespace)
-			}
-			backend, err := kongstate.NewServiceBackendForService(
-				k8stypes.NamespacedName{
-					Namespace: namespace,
-					Name:      string(backendRef.Name),
-				},
-				kongstate.PortDef{
-					Mode:   kongstate.PortModeByNumber,
-					Number: port,
-				},
-			)
-			if err != nil {
-				logger.Error(err, "failed to create ServiceBackend for backendRef")
-			}
-			if backendRef.Weight != nil {
-				backend.SetWeight(*backendRef.Weight)
+		nn := client.ObjectKey{
+			Name:      string(backendRef.Name),
+			Namespace: route.GetNamespace(),
+		}
+		if backendRef.Namespace != nil {
+			nn.Namespace = string(*backendRef.Namespace)
+		}
+
+		if backendRef.Kind == nil {
+			logger.Error(nil, "Object requested backendRef to target, but no Kind was specified, skipping...")
+			continue
+		}
+
+		var err error
+		switch *backendRef.Kind {
+		case "Service":
+			_, err = storer.GetService(nn.Namespace, nn.Name)
+		default:
+			err = fmt.Errorf("unsupported kind %q, only 'Service' is Supported", *backendRef.Kind)
+		}
+		if err != nil {
+			switch errors.As(err, &store.NotFoundError{}) {
```

Comment:
Couldn't this condition (and the above one as well) be simply an `if` statement? 🙂

Reply:
The one checking `backendRef.Kind` feels easily expandable this way in case we'd like to do so. The one here needs to check for both conditions: `err != nil` and the error type check. Do you have a particular proposal for this? I'm open to suggestions

Reply:
I meant something like below, nothing crucial as it's logically the same. 
```go
		var err error
		if *backendRef.Kind == "Service" {
			_, err = storer.GetService(nn.Namespace, nn.Name)
		} else {
			err = fmt.Errorf("unsupported kind %q, only 'Service' is Supported", *backendRef.Kind)
		}
		if err != nil {
			if errors.As(err, &store.NotFoundError{}) {
				logger.Error(nil, "Object requested backendRef to target, but it does not exist, skipping...")
			} else {
				logger.Error(nil, "Object requested backendRef to target, an error occurred, skipping...")
			}
			continue
		}
```

Reply:
I've changed the second `switch` into an `if`. PTAL now